Sources/Datadog/RUM/RUMMonitor/RUMCommand.swift

```diff
@@ -6,60 +6,97 @@
-    case addUserAction(userAction: RUMUserAction, attributes: [AttributeKey: AttributeValue], time: Date)
-
-    // MARK: - Commands Published by `RUMScopes`
-
-    /// Replaces `.startView` command for the first View started in the application.
-    case startInitialView(id: AnyObject, attributes: [AttributeKey: AttributeValue], time: Date)
-
-    // MARK: - Properties
-
-    /// Time of the command issue.
-    var time: Date {
-        switch self {
-        case .startView(_, _, let time),
-             .stopView(_, _, let time),
-             .addCurrentViewError(_, _, _, let time),
-             .startResource(_, _, let time),
-             .stopResource(_, _, let time),
-             .stopResourceWithError(_, _, _, let time),
-             .startUserAction(_, _, let time),
-             .stopUserAction(_, _, let time),
-             .addUserAction(_, _, let time),
-             .startInitialView(_, _, let time):
-            return time
-        }
-    }
-
+/// Command processed through the tree of `RUMScopes`.
+internal protocol RUMCommand {
+    /// The time of command issue.
+    var time: Date { get }
     /// Attributes associated with the command.
-    var attributes: [AttributeKey: AttributeValue] {
-        switch self {
-        case .startView(_, let attributes, _),
-             .stopView(_, let attributes, _),
-             .addCurrentViewError(_, _, let attributes, _),
-             .startResource(_, let attributes, _),
-             .stopResource(_, let attributes, _),
-             .stopResourceWithError(_, _, let attributes, _),
-             .startUserAction(_, let attributes, _),
-             .stopUserAction(_, let attributes, _),
-             .addUserAction(_, let attributes, _),
-             .startInitialView(_, let attributes, _):
-            return attributes
-        }
-    }
+    var attributes: [AttributeKey: AttributeValue] { get }
+}
+
+// MARK: - RUM View related commands
+
+internal struct RUMStartViewCommand: RUMCommand {
+    let time: Date
+    let attributes: [AttributeKey: AttributeValue]
+
+    /// The object (typically `UIViewController`) identifying the RUM View.
+    let identity: AnyObject
+
+    /// Used to indicate if this command starts the very first View in the app.
+    /// * default `false` means _it's not yet known_,
+    /// * it can be set to `true` by the `RUMApplicationScope` which tracks this state.
+    var isInitialView = false
+}
+
+internal struct RUMStopViewCommand: RUMCommand {
+    let time: Date
+    let attributes: [AttributeKey: AttributeValue]
+
+    /// The object (typically `UIViewController`) identifying the RUM View.
+    let identity: AnyObject
+}
+
+internal struct RUMAddCurrentViewErrorCommand: RUMCommand {
+    let time: Date
+    let attributes: [AttributeKey: AttributeValue]
+
+    /// The error message.
+    let message: String
+    /// The error object.
+    let error: Error?
+}
```

Comment:
(first, sorry for the long comment!)
we can do the following only if we can make a change here: https://github.com/DataDog/dd-sdk-ios/pull/190#discussion_r459485890
if we can't, you can ignore the rest of this comment 👌 

it looks like the main idea is that every scope is associated with a group of commands (eg: resource commands, action commands, view commands):
```swift
// ViewScope
process(cmd) {
  switch cmd {
    case is ViewCmd: applySideEffect(cmd)
    case is ResourceCmd: resourceScopes.forEach { $0.process(cmd as ResourceCmd) }
    case is ActionCmd: actionScopes.forEach { $0.process(cmd as ActionCmd) }
  }
}
// ResourceScope
process(resourceCmd) {
  switch resourceCmd.kind {
    case .start(let containerTupleIfThereAreTooManyParamsHere): doSmth(with: container)
    case .stop...
}
// RUMCommands: 1 type per group
struct RUMResourceCmd {
  enum Kind { .start, .stop, .error }
  date: Date
  attr: Attr
}
```
so that we need to test `RUMResourceScope` only with every config of `RUMResourceCommand`, not with other groups.
does that make sense?

Reply:
summary of zoom call: we keep things quite a bit unbounded as we are not yet sure how things will play out, so we keep `process(cmd: RUMCommand)` instead of `process(cmd: RUMResourceCommand)` in `ResourceScope`